packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -632,13 +634,15 @@ export class BaseResolversVisitor<
       applyWrapper: type => this.applyResolverTypeWrapper(type),
       clearWrapper: type => this.clearResolverTypeWrapper(type),
       getTypeToUse: name => this.getTypeToUse(name),
+      currentType: 'ResolversTypes',
```

Comment:
We will be using `currentType` as a generic so we don't have to create two types like `ResolversUnionTypes` and `ResolversUnionParentTypes`

I'll follow this PR up with a cleanup PR to clean up the Union counterpart